test/k8sT/Services.go

```diff
@@ -412,6 +412,139 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 	})
 
+	SkipContextIf(func() bool { return !helpers.RunsOnNetNextOr419Kernel() }, "Checks local redirect policy", func() {
+		const (
+			lrpServiceName = "lrp-demo-service"
+			fe1Name        = "k8s1-client"
+			fe2Name        = "k8s2-client"
+			be1Name        = "k8s1-backend"
+			be2Name        = "k8s2-backend"
+			repeat         = 10
+		)
+
+		var (
+			deploymentYAML string
+			ctx            context.Context
+			ctxCancel      context.CancelFunc
+			clusterIP      string
+			curlTCP        string
+			curlUDP        string
+		)
+
+		BeforeAll(func() {
+			deploymentYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-test.yaml")
+			res := kubectl.ApplyDefault(deploymentYAML)
+			res.ExpectSuccess("Unable to apply %s", deploymentYAML)
+			pods := []string{fe1Name, fe2Name, be1Name, be2Name}
+			for _, pod := range pods {
+				err := kubectl.WaitforPods(helpers.DefaultNamespace, fmt.Sprintf("-l %s", pod), helpers.HelperTimeout)
+				ExpectWithOffset(1, err).Should(BeNil())
+			}
+			ctx, ctxCancel = context.WithTimeout(context.Background(), 1*time.Second)
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, lrpServiceName)
+			Expect(err).To(BeNil(), "Cannot get svc IP")
+			httpSVCURL := fmt.Sprintf("http://%s/", clusterIP)
+			tftpSVCURL := fmt.Sprintf("tfpt://%s/hello", clusterIP)
+			curlTCP = helpers.CurlFailNoStats(httpSVCURL)
+			curlUDP = helpers.CurlFailNoStats(tftpSVCURL)
+		})
+
+		AfterAll(func() {
+			ctxCancel()
+		})
+
+		It("LRP connectivity", func() {
+			// Basic sanity check
+			ciliumPods, err := kubectl.GetCiliumPods()
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+			for _, pod := range ciliumPods {
+				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, fmt.Sprintf("cilium service list | grep %s", clusterIP), "Cannot retrieve services on cilium pod")
+				service.ExpectContains("LocalRedirect", "LocalRedirect is not present in the cilium service list")
+			}
+
+			By("Checking traffic goes to local backend")
+			testCases := []struct {
+				selector string
+				pod      string
+				want     string
+				notWant  string
+			}{
+				{
+					selector: "id=app1",
+					pod:      fe1Name,
+					want:     be1Name,
+					notWant:  be2Name,
+				},
+				{
+					selector: "id=app2",
+					pod:      fe2Name,
+					want:     be2Name,
+					notWant:  be1Name,
+				},
+			}
+			for _, tc := range testCases {
+				try := 0
+				for try < repeat {
+					res, err := kubectl.ExecInPods(ctx, helpers.DefaultNamespace, tc.selector, curlTCP)
+					Expect(err).To(BeNil(), "http curl failed in %s pod", tc.selector)
+					res[tc.pod].ExpectContains(tc.want)
```

Comment:
Nit: small comment saying that you are expecting to see pod name in the returned Hostname field would be useful.

Reply:
Done.